### PR TITLE
Add draft for readme documentation

### LINKED ISSUE #12: Document syntax and functionality
Put it in the README to begin with

### PR DESCRIPTION
**Pull request by tohanss**
Fix #12

### CODE REVIEW

**Comment by tohanss** [comment]
Maybe not a fix for the issue, could be something else?

File: README.md [hunk 1]
```diff
@@ -1,5 +1,146 @@
...
+        // Assert
+        assertThat(emptyStackTop, equalTo(value));
+        assertThat(stackTop, equalTo(value));
+REPOBEE-SANITIZER-END
+    }
+}
+```
+>Example 1: For this .java test file, the `santize-file` command will identify the START and END markers, and proceed to remove the code inbetween.
+
+`repobee-sanitizer` also supports the `REPOBEE-SANITIZER-REPLACE-WITH` marker. By adding a replace marker, we can specify code that should replace the removed code. Example as follows:
```
Comment (slarse) [inline README.md]: This use case example is incomplete, it doesn't show the result.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,146 @@
...
+````
+class StackTest {
+    @Test
+    public void topIsLastPushedValue() {
+        fail("Not implemented");
+    }
+}
+````
+
+> Example 3: Skeletonized code that is provided to students.
```
Comment (slarse) [inline README.md:93]:
```suggestion
> Example 3: Sanitized code that is provided to students.
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,146 @@
...
+
+We can see that the only code that remains inside the function is that of the `REPLACE-WITH` marker. This gives us the main usage that `repobee-santizer` was developed for, it allows us to combine finished solutions with the "skeletonized" solutions that are provided to students.
+
+# Prefixing
+Sometimes (usually) we want code that can run, its a good thing then that `repobee-sanitizer` blocks can be commented out! Example two will produce the same output as the following:
```
Comment (slarse) [inline README.md:98]:
```suggestion
Sometimes (usually) we want code that can run, its a good thing then that `repobee-sanitizer` blocks can be commented out! Example 2 will produce the same output as the following:
```

File: README.md [hunk 4]
```diff
@@ -1,5 +1,146 @@
...
+`repobee-sanitizer`:
+* Determines prefix as any text that comes before `REPOBEE-SANITIZER-BLOCK`
+* Only determines prefix on a block-to-block basis, meaning that the prefix selected at a `BLOCK` marker must be used untill and including the next `END` marker 
+    * This means that all `repobee-sanitizer` blocks can have individuall prefixes
+
+
+# Syntax
+for `repobee-sanitizer` to work, marker syntax must be correct, this includes spelling of the markers themselves, the markers currently are as follows:
+
+- REPOBEE-SANITIZER-BLOCK
```
Comment (slarse) [inline README.md:140]: Looking at this, I think we should probably rename the start marker to `REPOBEE-SANITIZER-START`. But let's leave that for later.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,146 @@
...
+    - Indicates the end of a block.
+
+If a marker is incorrectly spelled, `repobee-sanitizer` will report an error. 
 
 # License
 See [LICENSE](LICENSE) for details.
+
+# Terminology
+Block
+Marker
```
Comment (slarse) [inline README.md:157]:
```suggestion
```
This is redundant for now, terminology may change as we move along and this section doesn't explain anything anyway. Remove.

File: README.md [hunk 6]
```diff
@@ -1,5 +1,146 @@
...
+
+[WIP]
+
+# Usage
+`repobee-sanitizer` is currently under development, it currently supports the `sanitize-file` and `sanitize-repo` commands. Because `repobee-sanitizer` is a plugin to `repobee`, these commands are executed by running repobee with:
+
+> repobee -p repobee-sanitizer sanitize-repo
+
+(is this right?)
+
```
Comment (slarse) [inline README.md:16]:
```suggestion
```
Remove for now, we don't have usage worked out yet.

**Pull request merged**